bazel-haskell-deps.bzl

```
@@ -121,52 +121,125 @@ haskell_library(
-    deps = '[":grpc", ":libgpr"]' if is_windows else '["@grpc_nix//:grpc_lib"]'
-    extra_targets = """
+    http_archive(
+        name = "grpc_haskell_core",
+        build_file_content = """
+load("@com_github_digital_asset_daml//bazel_tools:fat_cc_library.bzl", "fat_cc_library")
+load("@com_github_digital_asset_daml//bazel_tools:haskell.bzl", "c2hs_suite")
+load("@rules_haskell//haskell:defs.bzl", "haskell_library")
+c2hs_suite(
+    name = "grpc-haskell-core",
+    srcs = [
+        "src/Network/GRPC/Unsafe/Constants.hsc",
+    ] + glob(["src/**/*.hs"]),
+    c2hs_src_strip_prefix = "src",
+    hackage_deps = ["clock", "managed", "base", "sorted-list", "bytestring", "containers", "stm", "transformers"],
+    c2hs_srcs = [
+        "src/Network/GRPC/Unsafe/Time.chs",
+        "src/Network/GRPC/Unsafe/ChannelArgs.chs",
+        "src/Network/GRPC/Unsafe/Slice.chs",
+        "src/Network/GRPC/Unsafe/ByteBuffer.chs",
+        "src/Network/GRPC/Unsafe/Metadata.chs",
+        "src/Network/GRPC/Unsafe/Op.chs",
+        "src/Network/GRPC/Unsafe.chs",
+        "src/Network/GRPC/Unsafe/Security.chs",
+    ],
+    compiler_flags = ["-XCPP", "-Wno-unused-imports", "-Wno-unused-record-wildcards"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":fat_cbits",
+    ],
+)
+
 fat_cc_library(
-  name = "grpc",
-  input_lib = "@com_github_grpc_grpc//:grpc",
+  name = "fat_cbits",
+  input_lib = "cbits",
 )
-# Cabal requires libgpr next to libgrpc. However, fat_cc_library of grpc
-# already contains gpr and providing a second copy would cause duplicate symbol
-# errors. Instead, we define an empty dummy libgpr.
-genrule(name = "gpr-source", outs = ["gpr.c"], cmd = "touch $(OUTS)")
-cc_library(name = "gpr", srcs = [":gpr-source"])
-cc_library(name = "libgpr", linkstatic = 1, srcs = [":gpr"])
-""" if is_windows else ""
+cc_library(
+  name = "cbits",
+  srcs = glob(["cbits/*.c"]),
+  hdrs = glob(["include/*.h"]),
+  includes = ["include/"],
+  deps = [
+    "@com_github_grpc_grpc//:grpc",
+  ]
+)
+""",
+        patch_args = ["-p1"],
+        patches = [
+            "@com_github_digital_asset_daml//bazel_tools:grpc-haskell-core-cpp-options.patch",
+        ],
+        sha256 = "531bbd4df2eca160be436074ade336a70cad3a6477df8d00d479440edfe9896b",
+        strip_prefix = "gRPC-haskell-0cb7999e9e89d0c17c5e1d917e97cc6e450b9346/core",
+        urls = ["https://github.com/awakesecurity/gRPC-haskell/archive/0cb7999e9e89d0c17c5e1d917e97cc6e450b9346.tar.gz"],
+    )
 
     http_archive(
-        name = "grpc_haskell_core",
+        name = "grpc_haskell",
+        build_file_content = """
+load("@rules_haskell//haskell:defs.bzl", "haskell_library")
+haskell_library(
+    name = "grpc-haskell",
+    srcs = glob(["src/**/*.hs"]),
+    deps = ["@grpc_haskell_core//:grpc-haskell-core", "@stackage//:base", "@stackage//:proto3-suite", "@stackage//:bytestring", "@stackage//:managed", "@stackage//:proto3-wire"],
```

Does `packages[...].deps` not work here?
```suggestion
    deps = packages["grpc-haskell"].deps,
```

---

probably does, I just wanted to write it out manually for debugging.